src/main/java/com/cleanroommc/groovyscript/gameobjects/GameObjectHandler.java

```diff
@@ -147,18 +205,35 @@ public Builder<T> addSignature(Class<?>... paramTypes) {
             return this;
         }
 
+        public Builder<T> documentation(String doc) {
+            this.documentation = doc;
+            return this;
+        }
+
+        public Builder<T> docOfType(String type) {
+            String mod = this.mod == null ? StringUtils.EMPTY : this.mod.getContainerName();
+            return documentation("returns a " + mod + " " + type);
+        }
+
         public void register() {
             if (this.name == null || this.name.isEmpty()) throw new IllegalArgumentException("Name must not be empty");
-            if (GameObjectHandlerManager.hasGameObjectHandler(this.name))
-                throw new IllegalArgumentException("GameObjectHandler with name " + this.name + " already exists");
-            if (this.mod != null && !Loader.isModLoaded(this.mod))
+            if (this.mod != null && !this.mod.isLoaded())
                 throw new IllegalArgumentException("Tried to register GameObjectHandler for mod " + this.mod + ", but it's not loaded");
             Objects.requireNonNull(this.handler, () -> "The GameObjectHandler function must no be null");
             Objects.requireNonNull(this.returnType, () -> "The GameObjectHandler return type must not be null");
             if (this.paramTypes.isEmpty()) this.paramTypes.add(new Class[]{String.class});
             if (this.defaultValue == null) this.defaultValue = () -> null;
-            GameObjectHandlerManager.registerGameObjectHandler(new GameObjectHandler<>(this.name, this.mod, this.handler, this.defaultValue,
-                                                                                       this.returnType, this.paramTypes, this.completer));
+            this.documentation = IDocumented.toJavaDoc(this.documentation);
+            GameObjectHandler<T> goh = new GameObjectHandler<>(this.name, this.mod, this.handler, this.defaultValue,
+                                                               this.returnType, this.paramTypes, this.completer, this.documentation);
+            GameObjectHandlerManager.registerGameObjectHandler(this.mod, goh);
+            if (this.mod != null) {
```

Comment:
why is this in `Builder#register` instead of `GameObjectHandlerManager.registerGameObjectHandler`?